src/lib/asn1/oid_maps.cpp

```diff
@@ -16,338 +13,1357 @@ namespace Botan {
 
 namespace {
 
-std::vector<std::pair<const char*, std::vector<uint32_t>>> oid_maps() {
+// The hash can collide so we must verify the actual value matches before returning
+std::optional<std::string> if_match(const OID& oid, std::initializer_list<uint32_t> val, const char* name) {
+   if(oid.matches(val)) {
+      return std::string(name);
+   } else {
+      return {};
+   }
+}
+
+std::optional<OID> if_match(std::string_view req, const char* actual, std::initializer_list<uint32_t> oid) {
+   if(req == std::string_view(actual)) {
+      return OID(oid);
+   } else {
+      return {};
+   }
+}
```

Comment:
This might as well take a string view right away (because string literals implicitly convert to it):

```suggestion
std::optional<OID> if_match(std::string_view req, std::string_view actual, std::initializer_list<uint32_t> oid) {
   if(req == actual) {
      return OID(oid);
   } else {
      return {};
   }
}
```